app/sidekiq/terms_of_use/sign_up_service_updater_job.rb

```diff
@@ -13,8 +13,23 @@ class SignUpServiceUpdaterJob
       attr_package_key = job['args'].first
       attrs = Sidekiq::AttrPackage.find(attr_package_key)
 
-      Rails.logger.warn('[TermsOfUse][SignUpServiceUpdaterJob] retries exhausted',
-                        { icn: attrs&.dig(:icn), exception_message: exception.message, attr_package_key: })
+      icn = attrs&.dig(:icn)
+      version = attrs&.dig(:version)
+      agreement = UserAccount.find_by(icn:)&.terms_of_use_agreements&.where(agreement_version: version)&.last
+      response = if agreement.present?
+                   agreement.accepted? ? 'accepted' : 'declined'
+                 end
```

Comment:
I think this is an example where the safe navigation operator might be useful. You could probably rewrite is as:

```ruby
response = agreement&.accepted? ? 'accepted' : 'declined'
```

You might also be able to do this instead:

```ruby
response = agreement.present? ? agreement.response : 'declined'
```

I think I like the second one more.

Reply:
i want it to be nil if it doesn't exist